Tests/SonarRunner.Shim.Tests/PropertiesWriterTest.cs

```diff
@@ -329,6 +350,30 @@ public void PropertiesWriter_GlobalSettingsWritten()
 
         #region Private methods
 
+        private void VerifyProjectBaseDir(string expectedValue, string teamBuildValue, string userValue, string[] projectPaths)
+        {
+            AnalysisConfig config = new AnalysisConfig();
+            PropertiesWriter writer = new PropertiesWriter(config);
+            config.SonarOutputDir = TestSonarqubeOutputDir;
+
+            config.SourcesDirectory = teamBuildValue;
+            config.SetConfigValue(SonarProperties.ProjectBaseDir, userValue);
+
+            using (new AssertIgnoreScope())
+            {
+                foreach (string projectPath in projectPaths)
```

Comment:
Actually WriteSettingsForProject does a bit more than just writing - it also records each project in an internal list. Flush uses this list to write out additional properties.